File: 2342. Max Sum of a Pair With Equal Sum of Digits/MaxSumofaPairWithEqualSumofDigits.java
import java.util.Arrays;

public class MaxSumofaPairWithEqualSumofDigits {
    public int numberSum(int x){
        int temp = 0;
        while (x != 0) {
            temp += x%10;
            x /= 10;
        }
        return temp;
    }

    public int maximumSum(int[] nums) {
        Arrays.sort(nums);
        int sol = -1, temp = 0;
        int[] numArr = new int[82];

        for(int i : nums){
            temp = numberSum(i);
            if (numArr[temp] != 0) {
                sol = Math.max(sol, numArr[temp]+i); 
            }
            numArr[temp] = i;

        }
        return sol;
    }
    public static void main(String[] args) {
        MaxSumofaPairWithEqualSumofDigits mDigits = new MaxSumofaPairWithEqualSumofDigits();
        System.out.println(mDigits.maximumSum(new int[]{18,43,36,13,7}));
    }
}

File: 2342. Max Sum of a Pair With Equal Sum of Digits/Max Sum of a Pair With Equal Sum of Digits.py

nums = [18,43,36,13,7]
sumList = [0 for _ in range(82)] 
nums.sort()
sol = 0

def numberSum(x):
    temp = 0
    for i in str(x):
        temp += int(i)
    return temp

for i in nums:
    temp = numberSum(i)
    if sumList[temp]:
        sol = max(sol, sumList[temp]+i)
    sumList[temp] = i

print(sol)